### ISSUE TITLE
"Can't read viper config" when invoking via `gitlab-ci.yml` (gitlab runner)

### ISSUE THREAD

**Issue by jbergstroem** [golangci/golangci-lint#190]
This is inside of a gitlab-runner (`brew install gitlab-runner`) using `golangci/golangci-lint:v1.9.3` with a pretty minimal config (`.gitlab-ci.yml`):
```yaml
go:
  image: golangci/golangci-lint:v1.9.3
  script:
    - golangci-lint run
```

`golangci-lint run` finishes successfully if I either run it myself from the docker image or if I run it from the host, but inside of the gitlab runner (test with above `gitlab-runner exec docker go`) I instead get:
```console
Running with gitlab-runner 11.0.2 (1db57d4a)
Using Docker executor with image golangci/golangci-lint:v1.9.3 ...
Pulling docker image golangci/golangci-lint:v1.9.3 ...
Using docker image sha256:bcd9030fdbdcbf914fb7d239f273c57617b30cecad9b8ca345205d6feafdc1d0 for golangci/golangci-lint:v1.9.3 ...
Running on runner--project-0-concurrent-0 via jess.local...
Cloning repository...
Cloning into '/builds/project-0'...
done.
Checking out bcdc2c99 as feature/go-ci...
Skipping Git submodules setup
level=error msg="Can't read config: can't read viper config: Unsupported Config Type \"\""
level=error msg="Can't read config: can't read viper config: Unsupported Config Type \"\""
ERROR: Job failed: exit code 3
FATAL: exit code 3                                 
```

The error probably lies with how the environment is initialized in the runner, but I felt I'd report it upstream if it is related to how golangci-lint uses viper.

**Comment by jirfag** [comment]
hi!
`golangci-lint run -v; ls -l`  in gitlab ci please

**Comment by jbergstroem** [comment]
@jirfag: If I override the entrypoint I seem to yield better success:
```yaml
go:
  image: 
    name: golangci/golangci-lint:v1.9.3
    entrypoint: [""]
  script:
    - golangci-lint run
```

Going to debug the gitlab runner (`gitlab-runner --debug exec docker go`) and see if I can find hints to why that's the case. Perhaps `CMD` is a better option in the Dockerfile?

**Comment by amenzhinsky** [comment]
Can we replace `ENTRYPOINT` with `CMD` in `Dockerfile`?
